diff --git i/x11-servers/xorg-server/Makefile w/x11-servers/xorg-server/Makefile
index d24217b0bb26..c6ac6adcd52b 100644
--- i/x11-servers/xorg-server/Makefile
+++ w/x11-servers/xorg-server/Makefile
@@ -23,9 +23,10 @@ SLAVE_PORT?=	no
 OPTIONS_SUB=	yes
 OPTIONS_DEFINE=	SUID
 OPTIONS_RADIO=	CONF
-OPTIONS_RADIO_CONF=	DEVD HAL
+OPTIONS_RADIO_CONF=	DEVD HAL UDEV
 DEVD_DESC=	Use devd for autoconfiguration of input devices
 HAL_DESC=	Use hald for autoconfiguration of input devices
+UDEV_DESC=	Use udev for autoconfiguration of input devices
 SUID_DESC=	Install the Xorg server with setuid bit set
 OPTIONS_DEFAULT=DEVD SUID
 
@@ -44,7 +45,7 @@ USE_XORG+=	bigreqsproto compositeproto damageproto dri2proto dri3proto \
 		xineramaproto xkbfile xproto xshmfence xtrans
 CONFIGURE_ARGS+=--without-doxygen --without-xmlto --without-fop \
 		--localstatedir=/var --with-shared-memory-dir=/tmp \
-		--disable-config-udev --disable-config-udev-kms \
+		--disable-config-udev-kms --disable-systemd-logind \
 		--without-dtrace --enable-glamor
 INSTALL_TARGET=	install-strip
 
@@ -84,6 +85,13 @@ CONFIGURE_ARGS+=	--enable-config-hal
 CONFIGURE_ARGS+=	--disable-config-hal
 .endif
 
+.if ${PORT_OPTIONS:MUDEV}
+LIB_DEPENDS+=	libudev.so:devel/libudev-devd
+CONFIGURE_ARGS+= --enable-config-udev
+.else
+CONFIGURE_ARGS+= --disable-config-udev
+.endif
+
 # We handle Xorg setuid in the plist. This allows to build xorg-server as a user.
 CONFIGURE_ARGS+=--disable-install-setuid
 
